common/app/com/commercetools/sunrise/common/controllers/SunriseFrameworkController.java

```diff
@@ -259,4 +263,12 @@ protected final void saveUnexpectedFormError(final Form<?> form, final Throwable
         form.reject("Something went wrong, please try again"); // TODO i18n
         logger.error("The CTP request raised an unexpected exception", throwable);
     }
+
+    protected final void overwriteCartInSession(final Cart cart) {
```

Comment:
I am not happy with putting it in the core, I would prefer that this is also solved via components which are called when a primary cart is loaded/updated/created